evcouplings/mutate/protocol.py

```diff
@@ -132,10 +138,147 @@ def standard(**kwargs):
+    Protocol:
+    Compare ECs for single proteins (or domains)
+    to 3D structure information
+
+    Parameters
+    ----------
+    Mandatory kwargs arguments:
+        See list below in code where calling check_required
+
+    Returns
+    -------
+    outcfg : dict
+        Output configuration of the pipeline, including
+        the following fields:
+
+        * mutation_matrix_file
+        * [mutation_dataset_predicted_file]
+    """
+    check_required(
+        kwargs,
+        [
+            "prefix", "model_file",
+            "mutation_dataset_file",
+            "segments"
+        ]
+    )
+
+    prefix = kwargs["prefix"]
+
+    outcfg = {
+        "mutation_matrix_file": prefix + "_single_mutant_matrix.csv",
+        "mutation_matrix_plot_files": [],
+    }
+
+    # make sure model file exists
+    verify_resources(
+        "Model parameter file does not exist",
+        kwargs["model_file"]
+    )
+
+    # make sure output directory exists
+    create_prefix_folders(prefix)
+
+    # load couplings object
+    first_segment = Segment.from_list(kwargs["segments"][0])
+    second_segment = Segment.from_list(kwargs["segments"][1])
+
+    c = MultiSegmentCouplingsModel(kwargs["model_file"], first_segment, second_segment)
+
+    # create the independent model
+    c0 = c.to_independent_model()
+
+    # create the inter-protein only Jij model
+    ci = c.to_inter_segment_model()
+
+    for model, type_ in [(c, "Epistatic"), (c0, "Independent"), (ci, "Inter_segment")]:
+        # interactive plot using bokeh
+        filename = prefix + "_{}_model".format(type_.lower(), )
+        output_file(
+            filename + ".html", "{} model".format(type_)
+        )
+        fig = evcouplings.visualize.mutations.plot_mutation_matrix(model, engine="bokeh")
+        save(fig)
+        outcfg["mutation_matrix_plot_files"].append(filename + ".html")
+
+        # static matplotlib plot
+        evcouplings.visualize.mutations.plot_mutation_matrix(model)
+        plt.savefig(filename + ".pdf", bbox_inches="tight")
+        outcfg["mutation_matrix_plot_files"].append(filename + ".pdf")
+
+    # create single mutation matrix table,
+    # add prediction by independent model and
+    # save to file
+    singles = single_mutant_matrix(
+        c, output_column="prediction_epistatic"
+    )
+
+    singles = predict_mutation_table(
+        c0, singles, "prediction_independent"
+    )
+
+    singles = predict_mutation_table(
+        ci, singles, "prediction_inter_segment"
+    )
+
+    singles.to_csv(outcfg["mutation_matrix_file"], index=False)
+
+    # Pymol scripts
+    outcfg["mutations_epistatic_pml_files"] = []
+    for model in ["epistatic", "independent", "inter_segment"]:
+        pml_filename = prefix + "_{}_model.pml".format(model)
+        evcouplings.visualize.mutations.mutation_complexes_pymol_script(
+            singles, pml_filename, effect_column="prediction_" + model,
+            segment_to_chain_mapping={"A_1": "A", "B_1": "B"}
```

Comment:
How about this:

For segment names, extract these directly from the input configuration like I did in compare/protocol for complexes

For chain names, make a global variable in compare/protocol that we can then import. We have to be consistent across protocols so that the .pml files can be used on renumbered pdbs from the compare stage of course.